tests/test_google_lifesciences.py

```diff
@@ -34,9 +34,14 @@ def cleanup_google_storage(prefix, bucket_name="snakemake-testing"):
     blobs = bucket.list_blobs(prefix="source")
     for blob in blobs:
         blob.delete()
-    # Using API we get an exception about bucket deletion
-    shell("gsutil -m rm -r gs://{bucket.name}/* || true")
-    bucket.delete()
+    blobs = bucket.list_blobs(prefix=prefix)
+    for blob in blobs:
+        if restrict_to is None or f"{bucket_name}/{blob.name}" in restrict_to:
```

Comment:
Should this be if `restrict_to` is not None AND the bucket path is in restrict_to? As is now, the default restrict_to being None doesn't make sense - we would always delete files.  Is that intended? And what is this variable saying? If so, I find the naming of the variable confusing.

Reply:
I would rename to something with `keep_` which is more logical to see if keep is None, then we want to remove everything. But then I still find the logic kind of confusing.

Reply:
Could we add that as a comment somewhere?